server/sockets/helpers/index.js

```diff
@@ -0,0 +1,48 @@
+const List = require('../../models/list.model');
+const { ListActionTypes } = require('../../common/variables');
+const { responseWithList } = require('../../common/utils');
+
+const updateListOnDashboardAndCohortView = (
+  socket,
+  listId,
+  dashboardClients,
+  cohortViewClients
+) => {
+  List.findOne({
+    _id: listId
+  })
+    .lean()
+    .exec()
+    .then(doc => {
+      const { viewersIds, cohortId } = doc;
+
+      if (dashboardClients.size > 0) {
+        viewersIds.forEach(id => {
+          const viewerId = id.toString();
+          const list = responseWithList(doc, id);
+
+          if (dashboardClients.has(viewerId)) {
+            // send to users that are on the dashboard view
+            socket.broadcast
+              .to(dashboardClients.get(viewerId))
+              .emit(ListActionTypes.CREATE_SUCCESS, list);
+          }
+        });
+      }
+
+      if (cohortId && cohortViewClients) {
```

Comment:
`cohortId && cohortViewClinets.size > 0`, cohortViewClients  is created in `socket/index.js` file so I'm not sure if this check will be falsy if there will be no users on cohort view. Like in arrays case you have to check the array's length or sth like `isEmpty(array)`.